### PR TITLE
Implemented handling of international characters

### PR DESCRIPTION
**Pull request by saur0n**
Implemented generation of glyphs which are not ASCII characters (characters of non-English languages, special characters, [box-drawing characters](https://en.wikipedia.org/wiki/Box-drawing_character) and so on).

Also, list of includes was sorted.

### CODE REVIEW

File: src/fontem.c [hunk 1]
```diff
@@ -283,7 +303,7 @@ void store_glyph(FT_Face *face, FT_GlyphSlotRec *glyph,
 
 	// Append to the post
 	char *str = malloc(len + 100);
-	snprintf(str, len + 100, "\t[%d] = &%s,  /* '%c' */\n", ch, gname, ch);
+	snprintf(str, len + 100, "\t[%u] = &%s,  /* '%s' */\n", ch, gname, mb(ch));
```
Comment (chrisy) [inline src/fontem.c:327]: This could produce a very large sparse array if the index contains wide characters; we really need hash table.

**Review by chrisy — COMMENTED** [review]
I was pondering this last week as it happens, thanks for reading my mind. 👍 

The big remaining issue, from my comment on fontem.c, is that the character table is indexed on the ordinal value; for a table with both ASCII-7 and 2-byte characters this can produce a very large sparse array. 

I can see two approaches, both involving a hash, there may be others:
* Compute a simple hash over a bucket sized for the number of characters present. Our indices are hash values and we include the hash calculator in fonrem.h
* Produce a packed array of characters, no index, and provide a function in fontem.h that will produce a hashed index at runtime.

First idea reduces startup delay in the runtime and if done correctly should produce a smaller array for most situations. The second idea makes sure the binary image is smallest, but requires runtime initialization and runtime memory for the hash which may not be available on all systems.

Anyway, I don't need this fixed in this patch, I just wanted to start a conversation about it.

[edited]  I deleted a comment about rebasing; the "merge from master" commit is actually an edit, weird.

**Comment by saur0n** [comment]
I see the following approach for the sparce array problem. Since all characters in the file are sorted by its code point, we can employ binary search for searching a particular character in the array. It is very fast: if there are 1024 characters in the font, only 10 simple steps is needed to find a character description (or to find out that there are no character with such code). This did not require adding hashes or other auxiliary info to the binary.

**Comment by chrisy** [comment]
Sounds good.

**Pull request merged**